kong/clustering/wrpc_control_plane.lua

```diff
@@ -173,6 +174,15 @@ _M.check_configuration_compatibility = clustering_utils.check_configuration_comp
 
 
 function _M:handle_cp_websocket()
+  if self.conf.legacy_hybrid_protocol then
+    ngx_log(ngx_DEBUG, "received a request to the wRPC listener, but wRPC is disabled")
+    return ngx_exit(405)
```

Comment:
The "legacy\_hybrid\_protocol" switch is barely an "escape hatch" to buy some time when things go wrong,  not intended for users, not documented intentionally and will be removed eventually.
I'm not sure if we should introduce 1 more status to let the client know the difference given the motivation for this switch.

Reply:
I can take this out, but I will need to add the `legacy_hybrid_protocol` gate to the [test template](https://github.com/Kong/kong/blob/8920006b2400bbc8e2ba8d329d8b6babc11aaf21/spec/fixtures/custom_nginx.template#L454-L458) that exists in the [main template](https://github.com/Kong/kong/blob/8920006b2400bbc8e2ba8d329d8b6babc11aaf21/kong/templates/nginx_kong.lua#L442-L448). If the tests pass with that in place it's good enough for me.

Reply:
> I can take this out, but I will need to add the `legacy_hybrid_protocol` gate to the [test template](https://github.com/Kong/kong/blob/8920006b2400bbc8e2ba8d329d8b6babc11aaf21/spec/fixtures/custom_nginx.template#L454-L458) that exists in the [main template](https://github.com/Kong/kong/blob/8920006b2400bbc8e2ba8d329d8b6babc11aaf21/kong/templates/nginx_kong.lua#L442-L448). If the tests pass with that in place it's good enough for me.

I think it's fine to handle the switch here(as returning 404 explicitly).